test/src/test/java/jenkins/security/BasicHeaderProcessorTest.java

```diff
@@ -155,4 +188,79 @@ public HttpResponse doIndex() {
             return HttpResponses.plainText(u!=null ? u.getId() : "anonymous");
         }
     }
+    
+    @TestExtension
+    public static class DummySecurityListener extends SecurityListener {
```

Comment:
For the future: Maybe it makes sense to move this to a separate utility class in https://github.com/jenkinsci/jenkins-test-harness/ later (and then just inherit the extension class for it). The logic may be useful in other security-relate test classes in plugins